### DISCUSSION TITLE
Copyright Headers?

### DISCUSSION THREAD

**Discussion by JulianFeinauer** [pragmaticindustries/open-machinery-plattform#16] [General]
First question, do we want / need copyright headers?
Second question, which one do we use?

I checked e.g. this post here https://softwareengineering.stackexchange.com/questions/124687/copyright-statements-in-source-code and would opt for headers.

Then, we need some derivative of e.g. this header

    Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at
    
      http://www.apache.org/licenses/LICENSE-2.0
    
    Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.    

Any suggestions or concerns?

**Comment by JulianFeinauer** [comment]
Good hint from @chrisdutz: https://github.com/conterra/mapapps-fontawesome/blob/master/apache2-license-header.txt

**Comment by NiklasMerz** [comment]
This stack exchange is ten years old. My only concern is that I personally don't like these headers as they bloat the files and come with the extra burden of checking them for every release. If we add them we are certainly safe.

I would like to better understand why they are needed because I never got a good answer. I get why it makes sense to include them in libraries or files delivered through browsers but for our backend code you mostly use the project as a whole with the license included.

**Reply by JulianFeinauer** [reply]
I think the stack exchange is pretty clear on that... someone could say he only looked at that file and didnt know there was a copyright on it and by placing it on every file you get that argument out of the way.
I think its fine if you set up your IDE once and it will handle all of that (I do it like that with Apache Headers)

**Reply by NiklasMerz** [reply]
The stack exchange talks mostly about copyright years. That's another point of discussion I think. How to handle these? Some projects seem to update them every year some don't.

I still don't think headers will stop users with bad intentions and users with good intentions will check the projects license.

If this is best practice then we should follow it. At least there is a [cool VSCode extension](https://marketplace.visualstudio.com/items?itemName=angelo-breuer.license-header-manager).

**Reply by JulianFeinauer** [reply]
I dont care that much and I think thats one of these cases where lawyers will say "It depends".
Pro Headers: It doesnt hurt, it gives legal "safety" when files are used out of context
Con: It adds a bit of cludder (its hidden for me in my IDE), we have to manage it (automatically add, check, ...)

For me this is fine but what do the others say?
